test/integration/setup/loadContracts.ts

```diff
@@ -12,19 +12,16 @@ interface MainnetContractsJSON {
 }
 
 export async function getAllContracts(): Promise<MainnetContracts> {
-  const contracts: MainnetContracts = {} as MainnetContracts;
   const addresses = mainnetAddresses as MainnetContractsJSON;
-
-  for (const mainnetAddressEntryName in addresses) {
-    const mainnetAddressEntry = addresses[mainnetAddressEntryName];
-    const artifactName = mainnetAddressEntry.artifactName;
-    const address = mainnetAddressEntry.address;
-    if (artifactName == 'unknown') continue;
-    const contract = await ethers.getContractAt(artifactName, address);
-    contracts[mainnetAddressEntryName] = contract;
-  }
-
-  return contracts;
+  const contractsAsArrayEntries = await Promise.all(
```

Comment:
I find this very hard to read and follow through what's happening, but it's an internal data structure creation routine so don't feel that strongly